lib/ansible/modules/cloud/amazon/ec2_vpc_nat_gateway.py

```diff
@@ -183,28 +192,40 @@
   returned: In all cases.
   type: str
   sample: [
-      {
-          'public_ip': '52.52.52.52',
-          'network_interface_id': 'eni-12345',
-          'private_ip': '10.0.0.100',
-          'allocation_id': 'eipalloc-12345'
-      }
+    {
+      'public_ip': '52.52.52.52',
+      'network_interface_id': 'eni-12345',
+      'private_ip': '10.0.0.100',
+      'allocation_id': 'eipalloc-12345'
+    }
   ]
+tags:
+  description: A dictionary of tags assigned to gateway
+  returned: In all cases.
+  type: dict
+  sample: {
+    "Env": "devel",
+    "Name": "nat-server"
+  }
 '''
 
 import datetime
 import random
 import time
+import traceback
 
 try:
     import botocore
 except ImportError:
-    pass  # caught by imported HAS_BOTO3
+    pass  # caught by imported ec2.HAS_BOTO3
 
+from ansible.module_utils._text import to_native
 from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.aws.core import AnsibleAWSModule
 from ansible.module_utils.ec2 import (ec2_argument_spec, get_aws_connection_info, boto3_conn,
```

Comment:
Once you replace the connection block below with module.client('ec2') you don't need these 3 imports...